src/Appwrite/Platform/Tasks/Backup.php

```diff
@@ -0,0 +1,252 @@
+        $time = date('Y_m_d_H_i_s');
+
+        self::log('--- Backup Start --- ');
+        self::log('--- Creating backup ' . $time . '  --- ');
+
+        $filename = $time . '.tar.gz';
+        $local = new Local(self::BACKUPS . '/' . $this->database . '/full/' . $time);
+        $local->setTransferChunkSize(5  * 1024 * 1024); // > 5MB
+
+        $backups = $local->getRoot() . '/files';
+        $tarFile = $local->getPath($filename);
+
+        $this->fullBackup($backups);
+        $this->tar($backups, $tarFile);
+        $this->upload($tarFile, $local);
+        // todo: Do we want to delete the tar file? and remain with the folder?
+        // todo: Do we want to delete the backup.log?
+
+        self::log('--- Backup End ' . (microtime(true) - $start) . ' seconds --- '   . PHP_EOL . PHP_EOL);
+    }
+
+    public function fullBackup(string $target)
+    {
+        if (!file_exists(self::BACKUPS)) {
+            Console::error('Mount directory does not exist');
+            Console::exit();
+        }
+
+        if (!file_exists($target) && !mkdir($target, 0755, true)) {
+            Console::error('Error creating directory: ' . $target);
+            Console::exit();
+        }
+
+        $logfile = $target . '/../backup.log';
+
+        $args = [
+            //'--defaults-file=' . $this->cnf, // [ERROR] Failed to open required defaults file: /etc/my.cnf
+            '--user=' . $this->dsn->getUser(),
+            '--password=' . $this->dsn->getPassword(),
+            '--host=' . $this->dsn->getHost(),
+            '--backup',
+            '--strict',
+            '--history=' . $this->database, // logs PERCONA_SCHEMA.xtrabackup_history name attribute
+            '--slave-info',
+            '--safe-slave-backup',
+            '--safe-slave-backup-timeout=300',
+            '--check-privileges', // checks if Percona XtraBackup has all the required privileges.
+            '--target-dir=' . $target,
+            '--parallel=' . self::PROCESSORS,
+            '--compress=' . self::COMPRESS_ALGORITHM,
+            '--compress-threads=' . self::PROCESSORS,
+            '--rsync', // https://docs.percona.com/percona-xtrabackup/8.0/accelerate-backup-process.html
+            //'--encrypt-threads=' . $this->processors,
+            //'--encrypt=AES256',
+            //'--encrypt-key-file=' . '/encryption_key_file',
+            //'--no-lock', // https://docs.percona.com/percona-xtrabackup/8.0/xtrabackup-option-reference.html#-no-lock
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec appwrite-xtrabackup xtrabackup ' . implode(' ', $args);
+        self::log($cmd);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+        Backup::log($stderr);
+
+        if (!str_contains($stderr, 'completed OK!') || !file_exists($target . '/xtrabackup_checkpoints')) {
+            Console::error('Backup failed');
+            Console::exit();
+        }
+
+        // todo: remove logfile?
```

Comment:
Let's remove this comment, we have a similar one elsewhere

Reply:
yep